File: sortapi.py
import sys
import time
import random

def doSort(arrlen):
  arr = [0]*arrlen

  for i in range(0, len(arr)):
    arr[i] = random.randrange(arrlen)
    if i % 2 == 0:
      arr[i] = -1 * arr[i]

  start = time.clock()
  arr.sort()
  diff = time.clock() - start
   
  for i in range(1, len(arr)):
    if arr[i-1] > arr[i]:
      print("not sorted")
      sys.exit()

  return diff

if len(sys.argv) != 2:
  print("usage: python sortapi.py ARRAYLENGTH")
  sys.exit()

alen = int(sys.argv[1])
min = 9999999999

for i in range(1, 6):
  t = doSort(alen)

  if t < min:
    min = t

print("python api sort: ", round(min*1000), " ms")

File: sortapi.rb
require "benchmark"

def doSort(len)
   arr = Array.new(len)

   for i in 0 ... arr.size
     arr[i] = Random.rand(len)
     if i % 2 == 0
       arr[i] = -1 * arr[i]
     end
   end

   ms = Benchmark.realtime do
     arr.sort!
   end
   
   for i in 1...arr.size
     if arr[i-1] > arr[i]
       abort("not sorted !!")
     end
   end

   ms * 1000
end

abort("Usage: ruby quicksort.rb ARRAYLENGTH") unless ARGV.length == 1

len = ARGV[0].to_i
min = 9999999999

for i in 1..5
  t = doSort(len)

  if t < min
    min = t
  end
end

puts "ruby quicksort: " + min.to_s + " ms"